src/amberscript/parser.cc

```diff
@@ -986,8 +992,50 @@ Result Parser::ParseRun() {
     if (!pipeline->IsGraphics())
       return Result("RUN command requires graphics pipeline");
 
+    token = tokenizer_->NextToken();
+    if (!token->IsString() || token->AsString() != "AS")
+      return Result("missing AS for RUN command");
+
+    token = tokenizer_->NextToken();
+    if (!token->IsString()) {
+      return Result("invalid topology for RUN command: " +
+                    token->ToOriginalString());
+    }
+
+    Topology topo = NameToTopology(token->AsString());
+    if (topo == Topology::kUnknown)
+      return Result("invalid topology for RUN command: " + token->AsString());
+
+    token = tokenizer_->NextToken();
+    if (!token->IsString() || token->AsString() != "START_IDX")
+      return Result("missing START_IDX for RUN command");
+
+    token = tokenizer_->NextToken();
+    if (!token->IsInteger()) {
+      return Result("invalid START_IDX value for RUN command: " +
+                    token->ToOriginalString());
+    }
+    if (token->AsInt32() < 0)
+      return Result("START_IDX value must be >= 0 for RUN command");
+    uint32_t start_idx = token->AsUint32();
+
+    token = tokenizer_->NextToken();
```

Comment:
This rejects the second DRAW_ARRAY case in the docs.

`RUN <pipeline_name> DRAW_ARRAY AS <topology> START_IDX <value>`

Reply:
https://github.com/google/amber/issues/474